src/providers/krb5/krb5_common.c

```diff
@@ -1141,38 +1141,29 @@ void krb5_finalize(struct tevent_context *ev,
                    void *siginfo,
                    void *private_data)
 {
+    struct be_ctx *be_ctx;
+
+    be_ctx = talloc_get_type(private_data, struct be_ctx);
+    if (be_ctx) {
```

Comment:
It is save to call `talloc_free()` with a `NULL` argument, so the `if` is not strictly needed. But I won't mind if you prefer to keep it for readability.

Reply:
Thank you, I was not aware of that. Code updated.